Userland/Libraries/LibCrypto/Hash/SHA1.cpp

```diff
@@ -7,18 +7,34 @@
 
 #include <AK/Endian.h>
 #include <AK/Memory.h>
+#include <AK/Platform.h>
 #include <AK/Types.h>
 #include <LibCrypto/Hash/SHA1.h>
 
+#if ARCH(I386) || ARCH(X86_64)
+#    include <emmintrin.h> // SSE2 _mm_add_epi32 _mm_loadu_si128 _mm_set_epi32 _mm_set_epi64x _mm_setzero_si128 _mm_shuffle_epi32 _mm_storeu_si128 _mm_xor_si128
+#    include <tmmintrin.h> // SSSE3 _mm_shuffle_epi8
+#    include <smmintrin.h> // SSE4.1 _mm_extract_epi32
+#    include <immintrin.h> // SHA _mm_sha1msg1_epu32 _mm_sha1msg2_epu32 _mm_sha1nexte_epu32 _mm_sha1rnds4_epu32
+#    define sha1_attribute_target_default __attribute__((target("default")))
+#    define sha1_attribute_target_x86 __attribute__((target("sse2,ssse3,sse4.1,sha")))
+#else
+#    define sha1_attribute_target_default
+#    define sha1_attribute_target_x86
+#endif
+
+
 namespace Crypto::Hash {
 
 static constexpr auto ROTATE_LEFT(u32 value, size_t bits)
 {
     return (value << bits) | (value >> (32 - bits));
 }
 
-inline void SHA1::transform(u8 const* data)
+sha1_attribute_target_default static void transform_impl(u32* state, u8 const* data)
```

Comment:
can we use Spans or Array& or something instead of raw pointers with no size information? This is partially pre-existing in this file but there's no reason to propagate that mistake further.

Reply:
> can we use Spans or Array& or something instead of raw pointers with no size information? This is partially pre-existing in this file but there's no reason to propagate that mistake further.

OK, will work on this.

Reply:
Fixed.